### PR TITLE
feat: HolderOf if someone used nft and then do resale

### LINKED ISSUE #8611: HolderOf if someone used nft and then do resale
If someone claims with motherboard nft Chroma collection and then resale that nft, it's not possible to use it again on Chroma?

just want to clear things how it's made rn

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8611

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-12-18 at 16 59 19@2x

## Copilot Summary
copilot:summary

copilot:poem

### CODE REVIEW

**Review by vikiival — COMMENTED** [review]
Does this actually solves anything?

**Comment by hassnian** [comment]
@vikiival going by what @exezbcz  mentioned https://github.com/kodadot/nft-gallery/issues/8611#issuecomment-1858837803 

> I would not add another action to the drop page claim. there is lot of chaos already

he said a ui warning will suffice

what other solution would you suggest

**Comment by exezbcz** [comment]
> > there is lot of chaos already

this, I can then design bit better solution

**Comment by vikiival** [comment]
Like the implementation is bad, basically you are checking if owner has nft in other collection eg (if someone with motherboard has chroma). 

so if I buy chroma now I would get warning :)

**Comment by vikiival** [comment]
If it helps as @exezbcz say feel free to merge.

**Comment by exezbcz** [comment]
> Like the implementation is bad, basically you are checking if owner has nft in other collection eg (if someone with motherboard has chroma).

then it should be different since it might happen that at least one NFT will be incorrectly labeled.

**Review by vikiival — CHANGES_REQUESTED** [review]
~

File: components/gallery/GalleryItemHolderOf.vue [hunk 1]
```diff
@@ -69,17 +59,9 @@ const showWarning = computed(
...
-
-    if (nFTEntities.length) {
-      alreadyClaimed.value = true
-    }
-  },
-  { immediate: true },
-)
+onBeforeMount(async () => {
+  alreadyClaimed.value = await checkIfAlreadyClaimed()
+})
```
Comment (vikiival) [inline components/gallery/GalleryItemHolderOf.vue:64]: API can be null at this point

File: components/gallery/GalleryItemHolderOf.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  <InfoBox v-if="showWarning" variant="warning">
+    <div class="is-flex is-align-items-center">
+      <NeoIcon icon="triangle-exclamation" class="mr-2" />
+      <span> {{ $t('drops.holderOfClaimed', [exclusiveDrop?.name]) }}</span>
+    </div>
+  </InfoBox>
+</template>
+
+<script setup lang="ts">
+import InfoBox from '@/components/shared/view/InfoBox.vue'
```
Comment (roiLeo) [inline components/gallery/GalleryItemHolderOf.vue:11]: auto imported
```suggestion
```

File: components/gallery/GalleryItem.vue [hunk 3]
```diff
@@ -140,6 +140,7 @@
               class="mt-4" />
 
             <!-- price section -->
+            <GalleryItemHolderOf v-if="nft" :nft="nft" />
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:143]: IMO this need to be only visible on assetHub

File: components/gallery/GalleryItemHolderOf.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const showWarning = computed(
+  () =>
+    isHolderOfCollection.value &&
+    alreadyClaimed.value &&
+    Boolean(exclusiveDrop) &&
+    !exclusiveDrop.disabled &&
+    hasAvailable.value,
+)
+
+onMounted(async () => {
```
Comment (roiLeo) [inline components/gallery/GalleryItemHolderOf.vue:62]: **suggestion**: `useAsyncData`

File: components/gallery/GalleryItemHolderOf.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  return wasUsed !== null
+}
+
+const showWarning = computed(
+  () =>
+    isHolderOfCollection.value &&
+    alreadyClaimed.value &&
+    Boolean(exclusiveDrop) &&
+    !exclusiveDrop.disabled &&
+    hasAvailable.value,
```
Comment (roiLeo) [inline components/gallery/GalleryItemHolderOf.vue:59]: wow many cases

File: components/drops/useDrops.ts [hunk 6]
```diff
@@ -21,6 +21,8 @@ export interface Drop {
   alias: string
 }
 
+export const DEFAULT_COLLECTION_MAX = 300
```
Comment (roiLeo) [inline components/drops/useDrops.ts:24]: move to `utils/drop`?

File: components/gallery/GalleryItem.vue [hunk 7]
```diff
@@ -194,6 +195,7 @@ import GalleryItemButton from './GalleryItemButton/GalleryItemButton.vue'
 import GalleryItemDescription from './GalleryItemDescription.vue'
 import GalleryItemTabsPanel from './GalleryItemTabsPanel/GalleryItemTabsPanel.vue'
 import GalleryItemAction from './GalleryItemAction/GalleryItemAction.vue'
+import GalleryItemHolderOf from './GalleryItemHolderOf.vue'
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:198]: auto imported
```suggestion
```

File: components/shared/view/InfoBox.vue [hunk 8]
```diff
@@ -44,6 +44,17 @@ const onClose = () => {
 
 $border_size: 1px;
 
+@mixin variantTheme($borderColor, $backgroundColor) {
```
Comment (roiLeo) [inline components/shared/view/InfoBox.vue:47]: any way to use tailwind instead?

**Review by roiLeo — COMMENTED** [review]
smoll

**Comment by yangwao** [comment]
do we have test cases to check?

**Comment by hassnian** [comment]
> do we have test cases to check?

no warning 

https://deploy-preview-8656--koda-canary.netlify.app/ahp/gallery/50-4


with warning

https://deploy-preview-8656--koda-canary.netlify.app/ahp/gallery/50-5
https://deploy-preview-8656--koda-canary.netlify.app/ahp/gallery/50-264

File: components/drops/useDrops.ts (hunk 6 above)
Comment (vikiival) [reply components/drops/useDrops.ts:24]: Default is infinity ;)

File: components/shared/view/InfoBox.vue (hunk 8 above)
Comment (hassnian) [reply components/shared/view/InfoBox.vue:47]: let me check

**Comment by yangwao** [comment]
> with warning

okey looks fair for me

File: utils/drop.ts [hunk 9]
```diff
@@ -1,5 +1,13 @@
 export const DEFAULT_DROP = '/ahp/drops/chroma'
 
+export const HOLDER_OF_DROP_MAP = {
+  '50': '51',
+}
+
```
Comment (vikiival) [inline utils/drop.ts:6]: This is obsolete then ?

File: utils/drop.ts [hunk 10]
```diff
@@ -15,3 +23,5 @@ export const AHP_GENERATIVE_DROPS = [
 ]
 
 export const AHP_POPULAR_DROP_COLLECTIONS = [...AHP_GENERATIVE_DROPS]
+
+export const DEFAULT_COLLECTION_MAX = 300
```
Comment (vikiival) [inline utils/drop.ts:27]: This is misleading :) 

I do not want any message from @yangwao telling me about 300 max bug. 

If max is null / 0 then it's unlimted

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by vikiival** [comment]
otherwise lgtm

File: utils/drop.ts (hunk 9 above)
Comment (hassnian) [reply utils/drop.ts:6]: it's used to check if the "exclusive drop" is live and can mint , makes no sense to show that message if the drop has ended

File: utils/drop.ts (hunk 10 above)
Comment (hassnian) [reply utils/drop.ts]: changed to FALLBACK_DROP_COLLECTION_MAX

**Comment by vikiival** [comment]
Still incorrect :) but well

Screenshot 2023-12-20 at 12 36 31

[69ac95f](https://github.com/kodadot/nft-gallery/pull/8656/commits/69ac95fdb5f0ac30cdc6f19cb42dd3fa7990a9f7)

**Review by vikiival — APPROVED** [review]

**Comment by hassnian** [comment]
> Still incorrect :) but well
> 
> Screenshot 2023-12-20 at 12 36 31
> 
> [69ac95f](https://github.com/kodadot/nft-gallery/pull/8656/commits/69ac95fdb5f0ac30cdc6f19cb42dd3fa7990a9f7)

this is my chnage 
CleanShot 2023-12-20 at 16 51 15@2x

you mean the number was already wrong ?

**Comment by prury** [comment]
warning message not showing 4 me? 
image

**Comment by yangwao** [comment]
Takes while tho 🤔

Works on mobile ✅

**Comment by hassnian** [comment]
> Takes while tho 🤔

about 2-3s for me , you mean more then that ?

https://github.com/kodadot/nft-gallery/assets/44554284/f1c99ba1-663a-4d96-a4ec-102b6228560a

**Comment by yangwao** [comment]
> you mean more then that ?

same

**Comment by prury** [comment]
> 
> about 2-3s for me , you mean more then that ?

for me its not showing at all: 

https://github.com/kodadot/nft-gallery/assets/36627808/c21f5d7f-0b70-4d93-9515-a15f7272842d

**Comment by hassnian** [comment]
> 
> 
> > 
> 
> > about 2-3s for me , you mean more then that ?
> 
> 
> 
> for me its not showing at all: 
> 
> 
> 
> https://github.com/kodadot/nft-gallery/assets/36627808/c21f5d7f-0b70-4d93-9515-a15f7272842d
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

logs ?

**Comment by prury** [comment]
> 
> any errors on the console

nope, but its magically working now, and i did nothing for it to be solved, nothing, haha
yesterday i cleared cache more than once, tried another browser, tried some other links

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 7.55 USD/DOT ~ 6.623 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xb555790176bd9d95f72986bed1e7dd60ac93bf3b19a567f2b60dcc7b84a3ffcc](https://polkadot.subscan.io/extrinsic/0xb555790176bd9d95f72986bed1e7dd60ac93bf3b19a567f2b60dcc7b84a3ffcc)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**